examples/typescript-example/src/App.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  useTable,
+  useGroupBy,
+  useFilters,
+  useSortBy,
+  useExpanded,
+  usePagination,
+  Column,
+  CellProps,
+  UseSortByColumnProps,
+  TableInstance,
+  UseFiltersColumnProps,
+  ColumnInstance,
+  useBlockLayout,
+  useResizeColumns,
+  UseResizeColumnsHeaderProps
+} from 'react-table'
+import styled from 'styled-components'
+import makeData from './makeData'
+
+type Data = object
+
+type Props = {
+  columns: Column<Data>[]
+  data: Data[]
+}
+
+interface TableColumn<D extends object = {}>
+  extends ColumnInstance<D>,
+    UseSortByColumnProps<D>,
+    UseResizeColumnsHeaderProps<D>,
+    UseFiltersColumnProps<D> {}
+
+const Rigth = styled.div`
+  text-align: right;
+`
+
+const OrderIndicator = styled.span<{ isSorted?: boolean; isSortedDesc?: boolean }>`
+  position: absolute;
+  right: 0;
+  left: 0;
+  height: 2px;
+  background: black;
+  ${({ isSorted, isSortedDesc }) => (isSorted ? (isSortedDesc ? { top: 0 } : { bottom: 0 }) : { display: 'none' })}
+`
+
+const TableWrap = styled.div`
+  display: inline-block;
+  border-spacing: 0;
+  border: 1px solid black;
+`
+
+const TableHead = styled.div``
+const TableBody = styled.div``
+const TableRow = styled.div``
+
+const TableCell = styled.div`
+  position: relative;
+  margin: 0;
+  padding: 0.5rem;
+  border-bottom: 1px solid black;
+  border-right: 1px solid black;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  :last-child {
+    border-right: 0;
+  }
+`
+const Resizer = styled.div`
+  display: inline-block;
+  width: 5px;
+  height: 100%;
+  position: absolute;
+  right: 0;
+  top: 0;
+  transform: translateX(50%);
+  z-index: 1;
+`
+
+const Input = styled.input`
+  width: calc(100% - 0.5rem);
+`
+
+const useStopPropagationOnClick = <T extends any>(ref: T) => {
+  useEffect(() => {
+    if (ref.current !== null) {
+      ref.current.addEventListener('click', (e: MouseEvent) => e.stopPropagation())
+    }
+  }, [])
+}
+
+const DefaultColumnFilter = ({
+  column: { filterValue, preFilteredRows, setFilter }
+}: {
+  column: TableColumn<Data>
+}) => {
+  const count = preFilteredRows.length
+  const inputRef = useRef<HTMLInputElement>(null)
+  useStopPropagationOnClick(inputRef)
+  return (
+    <Input
+      ref={inputRef}
+      value={filterValue || ''}
+      onChange={e => {
+        setFilter(e.target.value || undefined) // Set undefined to remove the filter entirely
+      }}
+      placeholder={`Search ${count} records...`}
+    />
+  )
+}
+
+const ResizerComponent: FC = props => {
+  const ref = useRef<HTMLDivElement>(null)
+  useStopPropagationOnClick(ref)
+  return <Resizer {...props} ref={ref} />
+}
+
+const defaultColumn = {
+  minWidth: 20,
+  width: 150,
+  maxWidth: 500,
+  Filter: DefaultColumnFilter
+}
+
+const Table: FC<Props> = ({ columns, data }) => {
+  // Use the state and functions returned from useTable to build your UI
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable<Data>(
```

Comment:
Did you get an error that Property 'getTableBodyProps' does not exist on type 'TableInstance<object>'? I'm getting this error and don't know how to work around this.

Reply:
@sophearychiv What version are you running?

Reply:
V7

Reply:
I'm pretty sure that the question was more along the lines of what beta of v7 are you using, but all that aside, you might alway want to take a look at https://github.com/tannerlinsley/react-table/pull/1534, and look at the Typescript.md file.

Reply:
oh, sorry! I'm using ^7.0.0-beta.1.

Reply:
I would suggest upgrading since a lot has changed. The types should be in-line with what the code has now on beta.12

Reply:
Thanks!